src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -114,4 +205,343 @@
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForDay(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            int hour = ((ZonedDateTime) map.get("day")).getHour();
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < 24; i++) {
+                if (hour == endDate.minusHours(i).getHour()) {
+                    result[endDate.getHour() - i] += amount;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType WEEK
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForWeek(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            ZonedDateTime date = (ZonedDateTime) map.get("day");
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < 7; i++) {
+                if (date.getDayOfMonth() == endDate.minusDays(i).getDayOfMonth()) {
+                    result[6 - i] += amount;
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType MONTH
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForMonth(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            ZonedDateTime date = (ZonedDateTime) map.get("day");
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < result.length; i++) {
+                if (date.getDayOfMonth() == endDate.minusDays(i).getDayOfMonth()) {
```

Comment:
I wonder whether there is no other mechanism to aggregate the data by day

Reply:
Improved the mechanism so we do not have to iterate through the possible spots